### ISSUE TITLE
NextCloud: No server data generated in nginx config

### ISSUE THREAD

**Issue by sjau** [NixOS/nixpkgs#95259]
**Describe the bug**
Recently, the NextCloud has been altered and nginx made only server support, see 128dbb31cca3ba479396c6b65946e2e6503c0f8d

However I tried to install NextCloud now but the current nextcloud.nix does not generate the appropriate entries for nginx.

**To Reproduce**
1. I have those entries in my configuration.nix:

```
    security.acme = {
        acceptTerms = true;
        email = "user@domain.tld";
    };

    # Enable NextCloud
    services.nextcloud = {
        enable = true;
        nginx.enable = true;
        home = "/data/nextcloud";
        hostName = "nc.domain.tld";
        https = true;
        logLevel = 2; # between 0 (Debug) and 4 (Fatal)
        autoUpdateApps = {
            enable = true;
            startAt = "03:00:00";
        };
        config = {
            overwriteProtocol = "https";
            dbtype = "pgsql";
            dbuser = "nextcloud";
            dbhost = "/run/postgresql"; # nextcloud will add /.s.PGSQL.5432 by itself
            dbname = "nextcloud";
            dbpassFile = "/data/nextcloud/dbpass";
            adminpassFile = "/data/nextcloud/adminpass";
            adminuser = "admin";
        };            
    };

    # Enable Nginx
    services.nginx = {
        enable = true;
        recommendedGzipSettings = true;
        recommendedOptimisation = true;
        recommendedProxySettings = true;
        recommendedTlsSettings = true;
        sslCiphers = "AES256+EECDH:AES256+EDH:!aNULL";
        virtualHosts = {
            "nc.domain.tld" = {
                addSSL = true;
                forceSSL = false;
                enableACME = true;
            };
        };
    };

    # Enable PostgreSQL
    services.postgresql = {
        enable = true;
        dataDir = "/data/postgres";
        ensureDatabases = [ "nextcloud" ];
        ensureUsers = [
            {   name = "nextcloud";
                ensurePermissions."DATABASE nextcloud" = "ALL PRIVILEGES";
            }
        ];
    };

    # Run PostGres before NextCloud
    systemd.services."nextcloud-setup" = {
        requires = ["postgresql.service"];
        after = ["postgresql.service"];
    };
```

2. I rebuilt the system and nextcloud didn't work.
3. Inspecting the nginx unit, it loads `             ├─2495 nginx: master process /nix/store/i81rp2hsv5lal5mbkwg1gjqxgs5kycp1-nginx-1.18.0/bin/nginx -c /nix/store/c6lcshrcqn0a59vxdwg5kr22v7fkfpy0-nginx.conf`

The config file contains:

```
pid /run/nginx/nginx.pid;
error_log stderr;
daemon off;
events {
}
http {
	# The mime type definitions included with nginx are very incomplete, so
	# we use a list of mime types from the mailcap package, which is also
	# used by most other Linux distributions by default.
	include /nix/store/r2wvgnr54vmwnjvzyqdixv8xbn362jgh-mailcap-2.1.48/etc/nginx/mime.types;
	include /nix/store/i81rp2hsv5lal5mbkwg1gjqxgs5kycp1-nginx-1.18.0/conf/fastcgi.conf;
	include /nix/store/i81rp2hsv5lal5mbkwg1gjqxgs5kycp1-nginx-1.18.0/conf/uwsgi_params;
	# optimisation
	sendfile on;
	tcp_nopush on;
	tcp_nodelay on;
	keepalive_timeout 65;
	types_hash_max_size 4096;
	ssl_protocols TLSv1.2 TLSv1.3;
	ssl_ciphers AES256+EECDH:AES256+EDH:!aNULL;
	# Keep in sync with https://ssl-config.mozilla.org/#server=nginx&config=intermediate
	ssl_session_timeout 1d;
	ssl_session_cache shared:SSL:10m;
	# Breaks forward secrecy: https://github.com/mozilla/server-side-tls/issues/135
	ssl_session_tickets off;
	# We don't enable insecure ciphers by default, so this allows
	# clients to pick the most performant, per https://github.com/mozilla/server-side-tls/issues/260
	ssl_prefer_server_ciphers off;
	# OCSP stapling
	ssl_stapling on;
	ssl_stapling_verify on;
	gzip on;
	gzip_proxied any;
	gzip_comp_level 5;
	gzip_types
	application/atom+xml
	application/javascript
	application/json
	application/xml
	application/xml+rss
	image/svg+xml
	text/css
	text/javascript
	text/plain
	text/xml;
	gzip_vary on;
	proxy_redirect          off;
	proxy_connect_timeout   90;
	proxy_send_timeout      90;
	proxy_read_timeout      90;
	proxy_http_version      1.0;
	include /nix/store/kxhk1rjxnsjx3cqfz2y8bjp7sc5ifdgr-nginx-recommended-proxy-headers.conf;
	# $connection_upgrade is used for websocket proxying
	map $http_upgrade $connection_upgrade {
		default upgrade;
		''      close;
	}
	client_max_body_size 10m;
	server_tokens off;
	server {
		listen 0.0.0.0:443 ssl http2 ;
		listen 0.0.0.0:80 ;
		server_name nc.domain.tld ;
		location /.well-known/acme-challenge {
			root /var/lib/acme/acme-challenge;
			auth_basic off;
		}
		ssl_certificate /var/lib/acme/nc.domain.tld/fullchain.pem;
		ssl_certificate_key /var/lib/acme/nc.domain.tld/key.pem;
		ssl_trusted_certificate /var/lib/acme/nc.domain.tld/full.pem;
	}
}
```

**Expected behavior**
It should have generated proper entries with the according location for nextcloud.

**Notify maintainers**
@schneefux @bachp @globin @fpletz @ma27

**Metadata**
Please run `nix-shell -p nix-info --run "nix-info -m"` and paste the result.
 - system: `"x86_64-linux"`
 - host os: `Linux 5.4.57, NixOS, 20.09pre237781.32b46dd897a (Nightingale)`
 - multi-user?: `yes`
 - sandbox: `yes`
 - version: `nix-env (Nix) 2.3.7`
 - channels(root): `"nixos-20.09pre237781.32b46dd897a"`
 - nixpkgs: `/nix/var/nix/profiles/per-user/root/channels/nixos`

Maintainer information:
```yaml
# a list of nixpkgs attributes affected by the problem
attribute:
# a list of nixos modules affected by the problem
module:
```

**Comment by Ma27** [comment]
I'm very sorry for this regression! I reproduced your problem locally and fixed the issue by reverting 128dbb31cca3ba479396c6b65946e2e6503c0f8d in fddeb7cb73096786de8e59a8502cf3ae523bd694.

**Comment by sjau** [comment]
That was quick :)
